cellrank/tl/kernels/_base_kernel.py

```diff
@@ -294,21 +295,37 @@ def compute_projection(
                 "Compute transition matrix first as `.compute_transition_matrix()`."
             )
 
+        if self.kernels[0]._conn is None:
+            raise AttributeError(
+                f"Connectivity matrix not defined for {self.kernels[0]}. As of now, "
+                "the embedding projection only works for kNN based kernels."
+            )
+
         start = logg.info(f"Projecting transition matrix onto `{basis}`")
         emb = _get_basis(self.adata, basis)
         T_emb = np.empty_like(emb)
 
+        T = self.transition_matrix.copy()
```

Comment:
Can you please rewrite this without using the copies? The code down below will be slightly more complicated,
but I think it's better (faster/memory-effiecient without copying).